### PR TITLE
Update types in label extension for clarity

### PR DESCRIPTION
**Pull request by jisantuc**
**Related Issue(s):** #

none

**Proposed Changes:**

1. Change `label:properties` in the label extension from a list of optional strings (`["abcde", null, "foo"]`) to an optional list of strings (`["abcde", "foo"]` or `null`)
2. Update class objects in labels from a list of strings or numbers (`["abcde", 34, "foo"]`) to a list of strings or a list of numbers (`["abcde", "foo"]` or `[0, 1, 2]`)

**PR Checklist:**

- [x] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- ~API only: I have run `npm run generate-all` to update the [generated OpenAPI files](https://github.com/radiantearth/stac-spec/blob/dev/api-spec/README.md#openapi-definitions).~

**Notes**

These changes come out of trying to interpret the types + words while working on azavea/stac4s#48

### CODE REVIEW

**Comment by m-mohr** [comment]
Hi! We can't merge this into master. Please redo the PR for dev. Also, there's a bit much space around the pipes. Other than that the PR looks good.

**Comment by m-mohr** [comment]
@jisantuc I just realized that the JSON Schema is not updated. `label:properties` doesn't allow `null` there and `classes` only allows strings. So either the readme or the schema needs to be changed so that both are aligned.

**Comment by jisantuc** [comment]
@m-mohr I think this is good for another look

File: extensions/label/json-schema/schema.json [hunk 1]
```diff
@@ -1,119 +1,135 @@
...
+  "id": "stac-label-item-schema.json#",
+  "title": "STAC Label Data Item Extension Spec",
+  "type": "object",
+  "properties": {
     "properties": {
+      "type": "object",
+      "required": [
+        "label:properties",
+        "label:classes",
+        "label:description"
```
Comment (m-mohr) [inline extensions/label/json-schema/schema.json:12]: label:type is missing

File: extensions/label/json-schema/schema.json [hunk 2]
```diff
@@ -1,119 +1,135 @@
...
-                    "type": "string"
+          ]
+        },
+        "label:classes": {
+          "title": "Classes",
+          "type": "array",
+          "items": {
+            "type": "object",
+            "required": [
+              "name",
```
Comment (m-mohr) [inline extensions/label/json-schema/schema.json:35]: The doc says name and classes aren't required. Either fix it here or in the docs.

File: extensions/label/json-schema/schema.json [hunk 3]
```diff
@@ -1,119 +1,135 @@
...
+        },
+        "label:methods": {
+          "title": "Method",
+          "type": "array",
+          "items": {
             "type": "string",
             "enum": [
-              "raster",
-              "vector"
+              "automated",
```
Comment (m-mohr) [inline extensions/label/json-schema/schema.json:88]: This doesn't follow the doc (highlighted text by me):

> Recommended to be a subset of 'automated' or 'manual', but **may be an arbitrary value.**

File: extensions/label/README.md [hunk 4]
```diff
@@ -59,10 +59,10 @@ Some additional notes are given here for some of the core STAC Item fields and w
 | label:overviews   | \[[Label Overview Object](#label-overview-object)] | Overview | An Object storing counts (for classification-type data) or summary statistics (for continuous numerical/regression data). |
 
 #### Class Object
-| Field Name | Type              | Name    | Description |
-| ---------- | ----------------- | ------- | ----------- |
-| name       | string\|null      | Name    | The property key within the asset's each `Feature` corresponding to class labels. If labels are raster-formatted, use null. |
-| classes    | \[string\|number] | Classes | The different possible class values within the property `name`. |
+| Field Name | Type                 | Name    | Description |
+| ---------- | -------------------- | ------- | ----------- |
+| name       | string\|null         | Name    | The property key within the asset's each `Feature` corresponding to class labels. If labels are raster-formatted, use null. |
```
Comment (m-mohr) [inline extensions/label/README.md:64]: Wouldn't it make sense to require something here? Same for the other objects below...

File: extensions/label/json-schema/schema.json (hunk 2 above)
Comment (matthewhanson) [reply extensions/label/json-schema/schema.json:35]: I don't think they should be required, so this should be fixed in the docs.

File: extensions/label/json-schema/schema.json (hunk 3 above)
Comment (matthewhanson) [reply extensions/label/json-schema/schema.json:88]: So I think the whole enum just needs to be removed then.

File: extensions/label/json-schema/schema.json (hunk 2 above)
Comment (jisantuc) [reply extensions/label/json-schema/schema.json]: @matthewhanson given your comment, I'm removing `required` here, though I'm not certain that's the right thing given that they are currently required here and not in the docs.

File: extensions/label/README.md (hunk 4 above)
Comment (m-mohr) [reply extensions/label/README.md:64]: Yeah, agreed. We should make a separate issue for that.

File: extensions/label/json-schema/schema.json (hunk 2 above)
Comment (m-mohr) [reply extensions/label/json-schema/schema.json]: I think the required should be in place and be added to the docs. How does an empty object makes any sense? Now you can specify [{}, {]] and it's valid?!

Comment (jisantuc) [reply extensions/label/json-schema/schema.json:36]: Required in both as of 04edb4c

File: extensions/label/README.md (hunk 4 above)
Comment (jisantuc) [reply extensions/label/README.md]: I see -- I misunderstood the relationship between your comments. I believe that 04edb4c addresses this as well.

**Pull request merged**